From 9dbb8992a4ee1155d25e654e680a6b671acd8dc8 Mon Sep 17 00:00:00 2001
From: Erwan Le Ray <erwan.leray@foss.st.com>
Date: Tue, 7 Apr 2020 11:26:09 +0200
Subject: [PATCH 108/529] serial: core: add support of rs485 delay in
 nanoseconds

Add support of rs485 nanoseconds delay to rs485conf structure.

With fast data rates (1Mbaut or more), 1ms become quite long. Users need
to be able to set shorter delays than 1 ms in order not to slow down the
channel unnecessarily. This enables faster communication channels without
changing the baudrate.

Signed-off-by: Erwan Le Ray <erwan.leray@foss.st.com>
---
 drivers/tty/serial/serial_core.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/drivers/tty/serial/serial_core.c
+++ b/drivers/tty/serial/serial_core.c
@@ -3420,6 +3420,16 @@ int uart_get_rs485_mode(struct uart_port
 	u32 rs485_delay[2];
 	int ret;
 
+	ret = device_property_read_u32_array(dev, "rs485-rts-delay-ns",
+					     rs485_delay, 2);
+	if (!ret) {
+		rs485conf->delay_rts_before_send_ns = rs485_delay[0];
+		rs485conf->delay_rts_after_send_ns = rs485_delay[1];
+	} else {
+		rs485conf->delay_rts_before_send_ns = 0;
+		rs485conf->delay_rts_after_send_ns = 0;
+	}
+
 	ret = device_property_read_u32_array(dev, "rs485-rts-delay",
 					     rs485_delay, 2);
 	if (!ret) {
